src/precompile.jl

```diff
@@ -0,0 +1,10 @@
+using SnoopPrecompile
+
+@precompile_all_calls begin
+    parser = Parser()
+    enable!(parser, FootnoteRule())
+    ast = parser("Hello *world*")
```

Comment:
Did you see whether more thorough syntax coverage here, rather than just italics, had a more noticeable effect, or did package load time increase significantly. Also would be good to add more `Rules` here to cover a broader range of syntax.

Reply:
This is before adding more syntax to precompilation: https://github.com/MichaelHatherly/CommonMark.jl/blob/04b27c2/benchmark/results/precompilation.md

Reply:
..and this is after: https://github.com/MichaelHatherly/CommonMark.jl/blob/ccd178e/benchmark/results/precompilation.md

[diff](https://github.com/MichaelHatherly/CommonMark.jl/commit/ccd178e58b04cbe2071a6391aeedbaf6da97657d#diff-aac2b970a542c4a6f4d4789dcb73839834b5451983fa7f5f031d7796b497d1fc)

Reply:
Close results. I'll use BenchmarkTools to eliminate noisy measurements.

Also note that the precompile times reported in those generated markdown files are not for the first run.

The firs precompile time of the most recent commit was somewhere around 10 seconds iirc

Reply:
Hm wait no BenchmarkTools makes no sense, it's about the first call.

To answer your question,
> Did you see whether more thorough syntax coverage here, rather than just italics, had a more noticeable effect, or did package load time increase significantly

Thorough syntax coverage seemed to have increased package load time by 0.08 seconds (11%).
For the first function calls:

Before (just italics precompiled):
`0.726420+1.245913+0.240084 = 2.212417` seconds

After (integration test precompiled):
`0.784270+1.239548+0.235873 = 2.259691` seconds

So these times _also_ increased slightly (by 0.05 seconds, or 2%).

Reply:
So, I'll go back to just the italics ig